exaudfclient/docs/script_options_design.md

```diff
@@ -0,0 +1,466 @@
+```
+{
+<option key A> => list(<option_value A.1>, <option_value A.2>,..., <option_value A.n>),
+<option key B> => list(<option_value B.1>, <option_value B.2>,..., <option_value B.m>)
+}
+```   
+
+
+
+Covers:
+- `req~multiple-options-management~1`
+- `req~duplicate-options-management~1`
+
+Tags: V2
+
+### Lexer and Parser Rules Whitespace Sequences
+`dsn~lexer-parser-rules-whitespace-sequences~1`
+
+Define the Lexer rules to tokenize whitespace sequences, and parser rules to recognize those white space characters as option separator.
+For white space character within an option value, those must be added as is to the result option value.
+For white space character in anything else, what is not an option, those white space characters must be ignored.
+
+
+
+Covers:
+- `req~white-spaces-script-options-parsing-v2~1`
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Lexer and Parser Rules Whitespace Escape Sequences
+`dsn~lexer-parser-rules-whitespace-escape-sequences~1`
+
+Define the Lexer rules to tokenize whitespace escape sequences:
+- '\ ' => space character
+- '\t' => <tab> character
+- '\f' => <form feed> character
+- '\v' => <vertical tab> character
+
+Implement rules which replace those white space escape tokens only at the beginning of an option value. Add parser rules to ignore those tokens in anything else, which is not a script option. Those token are not expected to be part of an option key.
+
+
+Covers:
+- `req~leading-white-spaces-script-options-parsing~1`
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Lexer and Parser Rules Escape Sequences
+`dsn~lexer-parser-rules-escape-sequences~1`
+
+Define the Lexer rules to tokenize '\n', '\r', '\; sequences, and parser rules to replace those sequences with <line feed>, <carriage return> or ';' characters. 
+Implement rules which replace those token at any location in an option value. Add parser rules to ignore those tokens in anything else, which is not a script option. Those token are not expected to be part of an option key.
+
+
+Covers:
+- `req~escape-sequence-script-options-parsing~1`
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Script Option Removal Mechanism
+`dsn~script-option-removal~1`
+
+Implement a method in class `ScriptOptionLinesParserCTPG` which removes *all* identified Script Options from the original script code at once. This method be executed after the import scripts are replaced. The algorithm must replace the script options in reverse order in order to maintain consistency of internal list of positions.
+
+
+Covers:
+- `req~script-option-removal~1`
+- `req~java-scriptclass-option-handling-v2~1`
+
+Tags: V2
+
+### Unknown Options Check
+`dsn~script-option-unknown-options-behvaior-v2~1`
+
+Implement a check in class `ScriptOptionLinesParserCTPG` which verifies that only known script options are found. Otherwise, raise an exception with information about the position and name of the unknown option.
```

Comment:
Hm, after a second thought this gets nasty, because the supported options depend on the handlers, but that a handler doesn't support certain options doesn't mean they are illegal, it is still possible that a subsequent handler can handle them. This means, we need centralize all supported options. This means also whenever we are adding a new option, we have to add it to the handler and the parser

Reply:
On the other hand, the Java parser is last handler in the chain. The Java parser does not need to know the DB parser keywords, because the DB implementation is supposed to remove all the options it knows.

It might become a problem if there are additional Script Options parser running after the Java, but I believe this is currently not possible.